pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -19,7 +19,7 @@ http_proxy=${http_proxy:-}
 
 # allow overwriting cacert's ca-bundle.crt with a custom one
 # this can be done by setting NIX_GIT_SSL_CAINFO and NIX_SSL_CERT_FILE environment variables for the nix-daemon
-GIT_SSL_CAINFO=${NIX_GIT_SSL_CAINFO:-$GIT_SSL_CAINFO}
+GIT_SSL_CAINFO=${NIX_GIT_SSL_CAINFO:-${NIX_SSL_CERT_FILE:-$GIT_SSL_CAINFO}}
```

Comment:
It's not obvious to me why we would prefer `NIX_SSL_CERT_FILE` over `GIT_SSL_CAINFO`. Can you add a comment explaining that decision?

Reply:
In fact, I'm now curious why we would have preferred `NIX_GIT_SSL_CAINFO` over `GIT_SSL_CAINFO` as well.

Reply:
I believe the confusion comes about because `fetchgit`'s `nix-prefetch-git` script runs in *two* different contexts:

It runs in Nix builds as a FO derivation. Here, `GIT_SSL_CAINFO` is a constant and designed to be a last fallback if none of the `impureEnvVars` versions override it.

However, it *also* runs as a standalone script for prefetching. In *this* context, `GIT_SSL_CAINFO` is *not* usually set. If it *is* set, it almost certainly isn't being set in a way where it's expected to be the fallback if two other environment variables are also not set.

I'm not sure if there's an obvious way to clear up this confusion. One way would be to bake in the default with a substitution so that it doesn't mess with expectations and make `GIT_SSL_CAINFO` only fallback to another value if it's not set.

Reply:
I'm starting to think this entire line should be moved out of `nix-prefetch-git` and into some sort of wrapper that only gets called by the Nix derivation. The problem is that `nix-prefetch-git`'s double-identity has been regularly forgotten and there's not a clear separation of "build-time" and "run-time" setup. If we clearly separate them, it may be easier to prevent this mistake into the future. The downside, of course, is that anything we do to this script has potential to break existing users/usage. There's probably little we can do to prevent that.